server/src/controllers/Preferences.ts

```diff
@@ -0,0 +1,208 @@
+        return res.json({ error: `Server failed with error ${error}` })
+    }
+};
+
+const setVisitedShare = async (req: Request, res: Response): Promise<Response> => {
+    const { userId, shareVisitedBars }: { userId: string, shareVisitedBars: boolean } = req.body;
+
+    // Check which params are missing
+    var handleEmpty: string = ''
+    handleEmpty = !userId ? 'userId' : '' || !shareVisitedBars ? 'shareVisitedBars' : ''; //find missing parm
+
+    if (handleEmpty) {
+        res.status(400);
+        return res.json({ error: `Failed to change visited share, missing field: ${handleEmpty}` });
+    };
+
+    // check if user exists
+    const user: any = await User.findOne({
+        where: {
+            id: userId
+        }
+    });
+
+    if (!user) {
+        res.status(400);
+        return res.json({error: "User not found" });
+    }
+
+    // Check if shareVisitedBars is not a boolean
+    if (typeof shareVisitedBars !== 'boolean') {
+        res.status(400);
+        return res.json({ error: `Failed to change visited share, 'shareVisitedBars' must be a boolean` });
+    };
+
+    try {
+        // Check if shareVisitedBars is already set to the desired value
+        const existingPreferences = await Preferences.findOne({ where: { userId: userId } });
+
+        if (existingPreferences && existingPreferences.shareLocation === shareVisitedBars) {
+            res.status(200);
+            return res.json({ success: `Share Visited Bars is already set to ${shareVisitedBars}` });
+        }
+
+        // Update shareLocation if preferences exist
+        if (existingPreferences) {
+            await Preferences.update({ shareVisitedBars: shareVisitedBars }, { where: { userId: userId } });
+            res.status(200);
+            return res.json({ success: `Share Visited Bars changed to ${shareVisitedBars}` });
+        } else {
+            res.status(400);
+            return res.json({ error: "Preferences entry not found for the user" });
+        }
+    } catch (error: any) {
+        res.status(500);
+        return res.json({ error: `Server failed with error ${error}` })
+    }
+};
+
+const setTimerSetting = async (req: Request, res: Response): Promise<Response> => {
+    const { userId, timerSetting }: { userId: string, timerSetting: number } = req.body;
+
+    // Check which params are missing
+    const handleEmpty: string = !userId ? 'userId' : !timerSetting ? 'timerSetting' : '';
+
+    if (handleEmpty) {
+        res.status(400);
+        return res.json({ error: `Failed to change timer setting, missing field: ${handleEmpty}` });
+    }
+
+    // Check if user exists
+    const user: any = await User.findOne({
+        where: { id: userId }
+    });
+
+    if (!user) {
+        res.status(400);
+        return res.json({ error: "User not found" });
+    }
+
+    // Check if timerSetting is not a number or is not within the valid range
+    if (typeof timerSetting !== 'number' || timerSetting < 1 || timerSetting > 180) {
```

Comment:
A time could be more than 180 in minutes. 180 minutes === 3hrs. It could also be 180 hrs === 7.5 days. Just check if its a whole number.

Reply:
Set the limit to be < Number.MAX_SAFE_INT